src/core/src/op/constant.cpp

```diff
@@ -117,28 +176,36 @@ Constant::Constant(bool memset_allocation, const element::Type& type, const Shap
 void Constant::allocate_buffer(bool memset_allocation) {
     // memset_allocation flag is to switch on initialization of objects in memory for element::string type
     // and set memory to zero for numeric element types
+    const auto num_elements = shape_size(m_shape);
+    const auto byte_size = element::get_byte_size(m_element_type, num_elements);
     if (m_element_type == ov::element::string) {
-        auto num_elements = shape_size(m_shape);
-        m_data = std::make_shared<StringAlignedBuffer>(num_elements, mem_size(), host_alignment(), memset_allocation);
+        m_data = std::make_shared<StringAlignedBuffer>(num_elements, byte_size, host_alignment(), memset_allocation);
     } else {
-        m_data = std::make_shared<AlignedBuffer>(mem_size(), host_alignment());
+        m_data = std::make_shared<AlignedBuffer>(byte_size, host_alignment());
         if (memset_allocation) {
             std::memset(m_data->get_ptr(), 0, m_data->size());
         }
     }
 }
 
 Constant::Constant(const element::Type& type, const Shape& shape, const void* data) : Constant(false, type, shape) {
+    const auto num_elements = shape_size(m_shape);
     if (m_element_type == ov::element::string) {
-        auto num_elements = shape_size(m_shape);
-        const std::string* src_strings = static_cast<const std::string*>(data);
-        std::string* dst_strings = static_cast<std::string*>(get_data_ptr_nc());
+        const auto src_strings = static_cast<const std::string*>(data);
+        const auto dst_strings = static_cast<std::string*>(get_data_ptr_nc());
         std::uninitialized_copy_n(src_strings, num_elements, dst_strings);
     } else {
-        std::memcpy(get_data_ptr_nc(), data, mem_size());
+        std::memcpy(get_data_ptr_nc(), data, element::get_byte_size(m_element_type, num_elements));
     }
 }
 
+Constant::Constant(const element::Type& type, const Shape& shape, const std::shared_ptr<ov::AlignedBuffer>& data)
+    : m_element_type(type),
+      m_shape(shape),
+      m_data{data} {
```

Comment:
Mixed style - `()` vs `{}`

Reply:
If there will any changes to this PR will change it or change with new PR where new types will be added.